### PR TITLE
[Serving] Enable sending model inputs as dict by new operation infer_dict or predict_dict

### PR DESCRIPTION
**Pull request by davesh0812**
https://jira.iguazeng.com/browse/ML-5705

### CODE REVIEW

File: tests/serving/test_serving.py [hunk 1]
```diff
@@ -529,6 +534,22 @@ def test_v2_get_modelmeta(rundb_mock):
...
 
+def test_v2_infer_dict(rundb_mock):
+    project = mlrun.new_project("tstsrv", save=False)
+    fn = mlrun.new_function("tst", kind="serving")
+    model_uri = _log_model(project)
+    print(model_uri)
+    fn.add_model("m1", model_uri, "ModelTestingClass", multiplier=100)
+
+    server = fn.to_mock_server()
+    resp_list = server.test("/v2/models/m1/infer", testdata_iris)
```
Comment (theSaarco) [inline tests/serving/test_serving.py:545]: Test also `/v2/models/m1/predict` - this will expose the issue I commented on earlier.

**Review by theSaarco — CHANGES_REQUESTED** [review]
I'm a bit concerned about our expanding the API which is supposed to be standard, but we're doing it anyway by introducing new types of inputs. So, at a minimum this should be documented.
Also, you have a small bug there - see comments.

File: mlrun/serving/v2_serving.py [hunk 2]
```diff
@@ -237,7 +239,12 @@ def do_event(self, event, *args, **kwargs):
         if not op and event.method != "GET":
             op = "infer"
 
-        if op == "predict" or op == "infer":
+        if (
+            op == "predict"
+            or op == "infer"
+            or op == "infer_dict"
+            or op == "predict_dict"
```
[the remark this answers is not in the document]
Comment (davesh0812) [reply mlrun/serving/v2_serving.py:246]: @theSaarco  You want me to open a documentation ticket for that?

[the remark this answers is not in the document]
Comment (theSaarco) [reply mlrun/serving/v2_serving.py:246]: Yes, I think we need to see where this makes sense to document - it should be part of the docstring, for sure, but maybe we want to introduce using this into one of the tutorials that demonstrate serving as well.

**Review by theSaarco — APPROVED** [review]
LGTM.

**Pull request merged**